### ISSUE TITLE
[Bug] Providing upgrade to "unknown" version

### ISSUE THREAD

**Issue by Cougar** [tomaae/homeassistant-mikrotik_router#240]
## Describe the issue
Somehow it can't read latest version info and then suggests to do an upgrade to "unknown" version

## Expected behavior
Find an actual latest version. It should also check if 6.x or 7.x version is running at the moment.

## Screenshots
![mikrotik](https://user-images.githubusercontent.com/60040/186529902-8372dfcc-6d6d-4304-a439-4dab258602e7.png)

## Software versions

 - Home Assistant version: 2022.8.6
 - Mikrotik Router integration version: v2.1.1
 - Mikrotik Hardware: 951G-2HnD
 - RouterOS version: 6.49.6

**Comment by tomaae** [comment]
I will need debugs for this.

**Comment by Cougar** [comment]
Redacted debug output is attached.

Strange is that `custom_components.mikrotik_router.version` is `0.0.0`.

`fw-update.status` is `ERROR: could not resolve dns name` which is actually correct because these CAP clients don't have internet access. CAPsMAN controller does upgrade them. It is probably not very common case and therefore not supported so far.

[config_entry-mikrotik_router-6b4a92317a5c1e4e48a77f4c04f93449.json.txt](https://github.com/tomaae/homeassistant-mikrotik_router/files/9435432/config_entry-mikrotik_router-6b4a92317a5c1e4e48a77f4c04f93449.json.txt)

**Comment by tomaae** [comment]
dont mind that version, that hardcoded version in manifest file.

this seems to be a shortcoming of the update integration. I have already contacted maintainer to find a standardized solution.

how is it setup that capsman controller upgrade slaves? I would like to detect such a feature, it does not make much sense to have an update sensor on slaves that have managed upgrades.

btw, those are not debugs, but diagnostics. but I understand the issue now so no need for debugs.

**Comment by Cougar** [comment]
It is not easy to detect unless you can somehow relate CAP clients with CAPsMAN. This is CAPsMAN (controller) setting
```
/caps-man manager upgrade-policy (none | require-same-version | suggest-same-upgrade)
```

**Comment by PavelD** [comment]
I had the same issue on the same model. 

When I tried to check updates on the web console I got DNS error or reported no internet.

Ping was working only for local IPs and not for the external one.

The problem has been resolved when i add default gateway as `/ip/route/add dst-address=0.0.0.0/0 gateway=192.168.1.1`.

Anyway It seems to me as not an issue. You can always hide/disable this generated sensor. And unknown version seems to  me as good value when checking is not working on the monitored system.

**Comment by tomaae** [comment]
> I had the same issue on the same model.
> 
> When I tried to check updates on the web console I got DNS error or reported no internet.
> 
> Ping was working only for local IPs and not for the external one.
> 
> The problem has been resolved when i add default gateway as `/ip/route/add dst-address=0.0.0.0/0 gateway=192.168.1.1`.
> 
> Anyway It seems to me as not an issue. You can always hide/disable this generated sensor. And unknown version seems to me as good value when checking is not working on the monitored system.

Its different root cause. For OP's case, update should not run like this due to centralized update system.

**Comment by sermayoral** [comment]
The bug still persists. 

Maybe its should control that an "unknown" version doesn't enable the update entity. Doesn't it? It's not the best solucion, but i think it's better than the actual behaviour.

**Comment by sermayoral** [comment]
Is not there a solution for this problem yet?? It's very annoying

**Comment by fusionstream** [comment]
@sermayoral Solution is here: https://github.com/tomaae/homeassistant-mikrotik_router/issues/463

@PavelD has a similar solution a few comments up sans the DNS settings

**Comment by Cougar** [comment]
This doesn't look like a solution but more like a conclusion :)

It would be a good idea to avoid `system/package/update/check-for-update` command. Should be much better to read MikroTik release URL directly from integration like discussed in https://forum.mikrotik.com/t/url-to-get-latest-versions-of-ros-branches/117002/8

**Comment by fusionstream** [comment]
Fair point. I'd still consider it a solution though as handling updates via the mikrotik's internal system takes care of package management for me.

Maybe, as per my original issue's request, it would be better with more descriptive errors? The user can then either set up the mikrotik correctly to allow for one click updates from HA, or do it the old way with uploading of packages.

**Comment by Cougar** [comment]
But the problem is not only the missing upgrade from HA (which I'm not sure, if this is a good idea), but the version check that doesn't detect available upgrades.

**Comment by fusionstream** [comment]
Yes. That's because it uses the status provided by the mikrotik itself; and if the mikrotik is not set up correctly, then it provides no acceptable answer.

Pulling update status by comparing an external source to the currently installed version will not allow you to update from HA itself or even from the router itself. That's like replacing tyres to give a better ride when really it's your suspension that's shot.

An expert will know how to get around it but it's still a bit of an eyesore. The best thing would be to fix the problem at its root. Or properly relay error information so that a human can step in and decide what to do.

**Comment by Cougar** [comment]
> Pulling update status by comparing an external source to the currently installed version will not allow you to update from HA itself or even from the router itself.

Of course, it’s best to initiate the upgrade directly from HA, but when that isn’t possible, it would still be helpful to show the user that an update is available. This is how Pi-hole handles it when upgrades from HA aren’t possible.

<img width="1075" height="846" alt="Image" src="https://github.com/user-attachments/assets/ca0017fd-6934-40f4-9ec8-29b4f0963156" />

<img width="966" height="578" alt="Image" src="https://github.com/user-attachments/assets/f541b0b8-8921-44be-b28a-767f12079570" />
